pkg/agent/init-container.go

```diff
@@ -10,6 +10,61 @@ import (
 	"k8s.io/apimachinery/pkg/api/resource"
 )
 
+func getFilePathCreationScript(configMap util.ConfigMap) ([]string, error) {
+
+	if configMap.Infisical.Auth.Type == util.KubernetesAuthType {
+
+		authConfig := util.KubernetesAuthConfig{}
+
+		if identityID, exists := configMap.Infisical.Auth.Config["identity-id"]; exists {
+			if id, ok := identityID.(string); ok {
+				authConfig.IdentityID = id
+			}
+		}
+
+		if authConfig.IdentityID == "" {
+			return []string{}, fmt.Errorf("identity-id is required for kubernetes auth")
+		}
+
+		return []string{
+			fmt.Sprintf("echo \"%s\" > %s/identity-id", authConfig.IdentityID, util.InitContainerAgentConfigVolumeMountPath),
+		}, nil
+	}
+
+	if configMap.Infisical.Auth.Type == util.LdapAuthType {
+
+		authConfig := util.LdapAuthConfig{}
+
+		if identityID, exists := configMap.Infisical.Auth.Config["identity-id"]; exists {
+			if id, ok := identityID.(string); ok {
+				authConfig.IdentityID = id
+			}
+		}
+		if username, exists := configMap.Infisical.Auth.Config["username"]; exists {
+			if username, ok := username.(string); ok {
+				authConfig.Username = username
+			}
+		}
+		if password, exists := configMap.Infisical.Auth.Config["password"]; exists {
+			if password, ok := password.(string); ok {
+				authConfig.Password = password
+			}
+		}
+
+		if authConfig.IdentityID == "" || authConfig.Username == "" || authConfig.Password == "" {
+			return []string{}, fmt.Errorf("identity-id, username, and password are required for ldap auth")
+		}
+
+		return []string{
+			fmt.Sprintf("echo \"%s\" > %s/username", authConfig.Username, util.InitContainerAgentConfigVolumeMountPath),
```

Comment:
looks like we aren't setting any file permissions? i think we should make them only readable by the owner? chmod 600